src/TestLoader.ts

```diff
@@ -78,14 +77,13 @@ class TestLoader {
     return this.environmentChangedEmitter.event;
   }
 
-  public async getTestState(forceReload: boolean = false): Promise<ProjectTestState> {
+  public async getTestState(forceReload = false): Promise<ProjectTestState> {
     if (forceReload) {
       if (this.promise) {
         // TODO handle if we are force reloading with an existing promise.  Need to cancel.
       }
 
       this.log.info(`Force loading all tests...`);
-      const self = this;
```

Comment:
Of all the changes in this PR, I think this one may alter the runtime behaviour.  The value of `this` should be the instance of `TestLoader`.  Maybe its better to create an external function to clear the `promise` field.

e.g.
```
const clearPromise = function() { this.promise = null; };
```